### PR TITLE
#3193 - New OCP Routes

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
## Main changes
- Added `certificate`, `key`, and `caCertificate` to routes.
- Added the proposed annotations to the same routes. Information about the route annotations and its intentions can be found in [route-specific-annotations_route-configuration](https://docs.openshift.com/container-platform/4.15/networking/routes/route-configuration.html#nw-route-specific-annotations_route-configuration).
- Renamed the routes (now sims.studentaidbc.ca) to allow both routes to exist at the same time for now.
- Only the certificate contents are provided as parameters to allow the correct line-breaking.

### Additional Steps
- Change Keycloak properties for: `aest`, `institution`, `student`, and `supporting-users`.
  -  Valid Redirect URIs: `https://dev.sims.studentaidbc.ca/*`
  - Web Origins: `https://dev.sims.studentaidbc.ca`
- Created a new GitHub variable `DOMAIN_PREFIX` (created as a regular var because it is not sensitive).
- Create the new variables `TLS_CERTIFICATE`, `TLS_KEY`, `TLS_CA_CERTIFICATE`. These were not added to the sample config because there is no reason for them to exist in the local environment.

### Non-directly related change
- `FORMIO_ROOT_EMAIL` and `FORMIO_ROOT_PASSWORD` are no longer regenerated at every form.io server deployment to avoid credentials reconfigurations along the applications. The credentials should be changed due to a password rotation process and not during the server update.

### TODO
- Ensure release instructions for the `Additional Steps` above.

### CODE REVIEW

File: devops/openshift/forms-deploy.yml [hunk 1]
```diff
@@ -216,9 +233,6 @@ parameters:
     description: FORMIO_ROOT_PASSWORD
     generate: expression
     from: "[a-zA-Z0-9]{16}"
```
Comment (dheepak-aot) [inline devops/openshift/forms-deploy.yml:235]: If this is not removed, will it supersede the assigned valued from agrument?

File: devops/openshift/queue-consumers-deploy.yml [hunk 2]
```diff
@@ -261,13 +261,30 @@ objects:
   - apiVersion: route.openshift.io/v1
     kind: Route
     metadata:
-      name: ${NAME}
+      name: ${NAME}-route
+      annotations:
+        haproxy.router.openshift.io/balance: leastconn
```
Comment (dheepak-aot) [inline devops/openshift/queue-consumers-deploy.yml:266]: 👍

File: devops/openshift/queue-consumers-deploy.yml [hunk 3]
```diff
@@ -261,13 +261,30 @@ objects:
...
+        haproxy.router.openshift.io/disable_cookies: "true"
+        haproxy.router.openshift.io/hsts_header: max-age=31536000;includeSubDomains;preload
+        haproxy.router.openshift.io/timeout: 30s
```
Comment (dheepak-aot) [inline devops/openshift/queue-consumers-deploy.yml:269]: Isn't this the default value?

image

File: devops/openshift/queue-consumers-deploy.yml [hunk 4]
```diff
@@ -261,13 +261,30 @@ objects:
...
         termination: edge
+        certificate: |
+          -----BEGIN CERTIFICATE-----
+          ${TLS_CERTIFICATE}
+          -----END CERTIFICATE-----
+        key: |
+          -----BEGIN PRIVATE KEY-----
+          ${TLS_KEY}
+          -----END PRIVATE KEY-----
+        caCertificate: |
```
Comment (dheepak-aot) [inline devops/openshift/queue-consumers-deploy.yml:284]: Is it not `destinationCACertificate`?? 

image

Comment (andrewsignori-aot) [reply devops/openshift/queue-consumers-deploy.yml:284]: The expected one is `caCertificate`. The [route](https://console.apps.silver.devops.gov.bc.ca/k8s/ns/0c27fb-dev/routes/tls-dev-edwapi/yaml) in the ticket can be used as an example.

File: devops/openshift/queue-consumers-deploy.yml (hunk 3 above)
Comment (andrewsignori-aot) [reply devops/openshift/queue-consumers-deploy.yml:269]: Followed the agreement in the ticket. I would be ok to have it removed.

File: devops/openshift/queue-consumers-deploy.yml (hunk 4 above)
Comment (dheepak-aot) [reply devops/openshift/queue-consumers-deploy.yml:284]: Ah!, I see now and thanks for explaining.

File: devops/openshift/forms-deploy.yml (hunk 1 above)
Comment (andrewsignori-aot) [reply devops/openshift/forms-deploy.yml]: Nice catch. I removed the auto-generated password.

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the changes. Looks good 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot

**Comment by andrewsignori-aot** [comment]
> @andrewsignori-aot
> 
> E2E test seems failed in this PR, it needs to pass all E2E test?

The failed tests are intermittent issues that also exist in other PRs. 
The ticket #3652 was created a few days ago to investigate those.

**Pull request merged**